app/components/HighlighterModule/highlighterModule.js

```diff
@@ -0,0 +1,437 @@
+var HighlighterModule = (function () {
+    var privateText = {},
+        privateTopics = [],
+        privateCurrentTopic = [],
+        privateDefColors = ['rgb(241, 96, 97)', 'rgb(253, 212, 132)', 'rgb(175, 215, 146)', 'rgb(168, 210, 191)', 'rgb(255,153,000)', 'rgb(102,000,153)', 'rgb(000,153,153)', 'rgb(255,102,255)', 'rgb(000,051,153)', 'rgb(153,000,204)', 'rgb(70,194,64)', 'rgb(94,242,188)'],
+        privateColors = [],
+        privateArticleIndex = [],
+        privateHighlights = [],
+        privateSelected = [],
+        privateContainer = 'no id specified';
+
+    /* ---------- Private Methods  --------- */
+
+    function mergeColors(topics, selected) {
+
+      var colors = privateColors;
+      var list = [];
+      var index = 0;
+      for (var current=0;current <topics.length;current+=1) {
+        for (var master=0;master <privateTopics.results.length;master+=1){
+          if (topics[current] == privateTopics.results[master].id) {
+            list.push(colors[master]);
+          }
+        }
+      }
+
+      var fraction = 1 / list.length;
+      var red = 0;
+      var blue = 0;
+      var green = 0;
+      index = 0;
+      while (index < list.length) {
+        var rgb = list[index].replace(/[^\d,]/g, '').split(',');
+        red += fraction * Number(rgb[0]);
+        green += fraction * Number(rgb[1]);
+        blue += fraction * Number(rgb[2]);
+        index += 1;
+      }
+      var opacity = 0.5;
+      if (selected) {
+        opacity = 1;
+      }
+      if (list.length == 0) {
+        return 'rgba(255, 255, 255, 0)';
+      }
+      return 'rgba(' + Math.round(red) + ', ' + Math.round(green) + ', ' + Math.round(blue) + ', ' + opacity +')';
+
+    }
+
+    function processHighlights() {
+      var highlights = privateHighlights;
+      var topics = privateTopics.results;
+      var selectedHighlights = privateSelected;
+      var parsedHighlights = [];
+      var final = [];
+      var temp_index = 0;
+      while (temp_index < highlights.length) {
+        var i = highlights[temp_index];
+        var start = {type: 'start', index: i.start, topic: i.topic, source: i, selected: false};
+        var end = {type: 'end', index: i.end, topic: i.topic, source: i, selected: false};
+        parsedHighlights.push(start);
```

Comment:
why are we not treating each highlight as a single object with a start and end property?